pennylane/templates/subroutines/qubitization.py

```diff
@@ -163,31 +141,16 @@ def compute_decomposition(*_, **kwargs):  # pylint: disable=arguments-differ
 
         **Example:**
 
-        >>> print(qml.Qubitization.compute_decomposition(hamiltonian = 0.1 * qml.Z(0), control = 1))
-        [AmplitudeEmbedding(array([1., 0.]), wires=[1]), Select(ops=(Z(0),), control=Wires([1])), Adjoint(AmplitudeEmbedding(array([1., 0.]), wires=[1])), Reflection(, wires=[0])]
-
+        >>> print(qml.Qubitization.compute_decomposition(hamiltonian = 0.1 * qml.Z(0), control = 1)
+        [Reflection(3.141592653589793, wires=[1]), PrepSelPrep(coeffs=(0.1,), ops=(Z(0),), control=Wires([1]))]
         """
 
         hamiltonian = kwargs["hamiltonian"]
         control = kwargs["control"]
 
-        coeffs, unitaries = _positive_coeffs_hamiltonian(hamiltonian)
-
         decomp_ops = []
 
-        decomp_ops.append(
-            qml.AmplitudeEmbedding(qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control)
-        )
-
-        decomp_ops.append(qml.Select(unitaries, control=control))
-        decomp_ops.append(
-            qml.adjoint(
-                qml.AmplitudeEmbedding(
-                    qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
-                )
-            )
-        )
-
         decomp_ops.append(qml.Reflection(qml.Identity(control)))
+        decomp_ops.append(qml.PrepSelPrep(hamiltonian, control=control))
```

Comment:
Why are we changing the order? 🤔

Reply:
In the literature it can be found in both ways. We have decided (approved by JM) to change the order to fit the explanation of the demo